gmso/formats/itp.py

```diff
@@ -0,0 +1,405 @@
+import unyt as u
+
+import gmso
+from gmso.core.angle import Angle
+from gmso.core.angle_type import AngleType
+from gmso.core.atom import Atom
+from gmso.core.atom_type import AtomType
+from gmso.core.bond import Bond
+from gmso.core.bond_type import BondType
+from gmso.core.dihedral import Dihedral
+from gmso.core.dihedral_type import DihedralType
+from gmso.core.improper import Improper
+from gmso.core.improper_type import ImproperType
+
+
+def read_itp(itp_file):
+    """Create a topology from a provided gro file.
+    The Gromos87 (gro) format is a common plain text structure file used
+    commonly with the GROMACS simulation engine.  This file contains the
+    simulation box parameters, number of atoms, the residue and atom number for
+    each atom, as well as their positions and velocities (velocity is
+    optional).  This only takes a string, to be passed into with open(filename,"r") and returns gmso `topology`.
+
+    Parameters
+    ----------
+    filename : str
+        The path to the gro file either as a string, or a file object that
+        points to the gro file.
+    Returns
+    -------
+    gmso.core.topology
+        A `topology` object containing site information
+    Notes
```

Comment:
Are there notes we should add? If not, let's remove this, if so then let's add them. Having some helpful notes is always better than not.